### PR TITLE
fix(settings): Localized components not wrapped in LocalizationProvider

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We were seeing <Localized /> component was not properly wrapped in a <LocalizationProvider /> errors in Sentry
* Locally, replicated the error in ResetPassword and CompleteResetPassword pages when errors occurred with callback refs
* The AppErrorDialog compoment was outside the LocalizationProvider, so this error could occur when a general application error is encountered
* Error tooltip on the PageDisplayName was showing unlocalized messages that weren't intended to be user facing

## This pull request

* Resolves errors with callback refs in CompleteResetPassword and ResetPassword
* Moves the LocalizationProvider outside the AppErrorBoundary to provide access to localization for the AppErrorDialog component
* Removes the PageDisplayName tooltip and replace with a more user-appropriate error message

## Issue that this pull request solves

Closes: #FXA-8557

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-settings/src/components/Settings/PageDisplayName/index.test.tsx [hunk 1]
```diff
@@ -103,24 +103,3 @@ it('displays a general error in the alert bar', async () => {
...
-    </AppContext.Provider>
-  );
-  // suppress the console output
-  let consoleErrorSpy = jest
-    .spyOn(console, 'error')
-    .mockImplementationOnce(() => {});
-  await submitDisplayName('Jane Nope');
-  expect(screen.getByTestId('tooltip')).toBeInTheDocument();
-  consoleErrorSpy.mockRestore();
-});
```
Comment (chenba) [inline packages/fxa-settings/src/components/Settings/PageDisplayName/index.test.tsx:126]: Deleted because we don't do this anymore?

Comment (chenba) [reply packages/fxa-settings/src/components/Settings/PageDisplayName/index.test.tsx:126]: Oh never mind I see the changes below.

File: packages/fxa-settings/src/lib/glean/index.ts [hunk 2]
```diff
@@ -158,7 +158,7 @@ export const GleanMetrics: Pick<
     // Starting with glean.js v2, accessing localStorage during
     // initialization could cause an error
     try {
-      if (config.enabled) {
+      if (config.enabled && !gleanEnabled) {
```
Comment (chenba) [inline packages/fxa-settings/src/lib/glean/index.ts:161]: Is this a related change?  I'm not sure why it is here.

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx [hunk 3]
```diff
@@ -84,44 +84,8 @@ const CompleteResetPassword = ({
...
-
-    const handleRecoveryKeyStatus = async () => {
-      if (!location.state?.lostRecoveryKey) {
-        await checkForRecoveryKeyAndNavigate(linkModel.email);
-      }
-      renderCompleteResetPassword();
-    };
-
-    const checkForRecoveryKeyAndNavigate = async (email: string) => {
+  const checkForRecoveryKeyAndNavigate = useCallback(
```
Comment (chenba) [inline packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx:87]: Did things just get moved around a bit in this file?  It looks like that in the diff but I'm not sure.

**Review by chenba — CHANGES_REQUESTED** [review]
Thanks for catching whatever's going on with the gql-api error handling in the display name component.

But I don't think the Glean change is necessary.  The `initialize` function should be called once and `gleanEnabled` is set in that function, so `initialize` should not be using the value of `gleanEnabled`.

File: packages/fxa-settings/src/components/Settings/PageDisplayName/index.test.tsx (hunk 1 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/components/Settings/PageDisplayName/index.test.tsx:126]: Yes, removed because we were directly showing any gql error that occurred. For example, I saw an ECONREFUSED error in the tooltip while testing other things. Ideally we might want to filter errors and show some in the tooltip, but the generic error should be sufficient for now. I might file an issue to more selectively show an error in the tooltip if the submitted name does not respect validation criteria.

File: packages/fxa-settings/src/lib/glean/index.ts (hunk 2 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/lib/glean/index.ts:161]: Agree, this is a stray change that is unrelated to this PR (sorry!). Was looking to resolve the `(Glean.core.Glean) Attempted to initialize Glean, but it has already been initialized. Ignoring.` alert that occurs on every load of settings. Will remove!

File: packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx (hunk 3 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx:87]: Yes, moved some functions out of the `useEffect` which reduced the number of required dependencies.

**Review by chenba — APPROVED** [review]
👍
